airflow/ui/src/pages/Run/TaskInstances.tsx

```diff
@@ -101,13 +180,59 @@ export const TaskInstances = () => {
         limit: pagination.pageSize,
         offset: pagination.pageIndex * pagination.pageSize,
         orderBy,
+        state: filteredState === null ? undefined : [filteredState],
+        taskDisplayNamePattern: Boolean(taskDisplayNamePattern)
+          ? taskDisplayNamePattern
+          : undefined,
       },
       undefined,
       { enabled: !isNaN(pagination.pageSize) },
     );
 
   return (
-    <Box>
+    <Box pt={4}>
+      <HStack>
+        <Select.Root
+          collection={stateOptions}
+          maxW="250px"
+          onValueChange={handleStateChange}
+          value={[filteredState ?? "all"]}
+        >
+          <Select.Trigger colorPalette="blue" isActive={Boolean(filteredState)}>
```

Comment:
Small nit, can you make this "clearable" like described in https://www.chakra-ui.com/docs/components/select#clear-trigger - then the filter on state can be reset with a single click
```suggestion
          <Select.Trigger colorPalette="blue" isActive={Boolean(filteredState)} clearable>
```

Reply:
Thanks, done in https://github.com/apache/airflow/pull/45215/commits/fe68b6f642ccdaa1b28486b96a5af8c306d0e3ad